### ISSUE TITLE
Cannnot import pydra.engine after pip install -e of a pydra task package

### ISSUE THREAD

**Issue by axiezai** [nipype/pydra-mrtrix3#2]
Hello, 

I tried re-installing everything in a fresh conda environment, what I did:
```
conda create -n pydra ipython
cd pydra
source activate pydra
pip install -e .[dev]
pip install pre-commit
pre-commit
cd ../pydra-mrtrix3/
pip install -e .
python -c "import pydra.engine"
```

Expected: the import should work because pydra is installed. 

Current behavior: python cannot find pydra.engine. Before doing `pip install -e .` in `pydra-mrtrix3` or any task repo like `pydra-ants`, the import works perfectly in the newly created conda environment. But after doing pip install of a task package, the import is ruined. 

My conda environment before installing the pydra-mrtrix3 package is as follows:
```
21:07 $ conda list
# packages in environment at /Users/xxie/miniconda3/envs/pydra:
#
# Name                    Version                   Build  Channel
apipkg                    1.5                      pypi_0    pypi
appdirs                   1.4.4                    pypi_0    pypi
appnope                   0.1.0                    py38_0
argparse                  1.4.0                    pypi_0    pypi
attrs                     19.3.0                   pypi_0    pypi
backcall                  0.2.0                      py_0
black                     19.10b0                  pypi_0    pypi
boutiques                 0.5.25                   pypi_0    pypi
ca-certificates           2020.1.1                      0
cachetools                4.1.0                    pypi_0    pypi
certifi                   2020.6.20                py38_0
cfgv                      3.1.0                    pypi_0    pypi
chardet                   3.0.4                    pypi_0    pypi
ci-info                   0.2.0                    pypi_0    pypi
click                     7.1.2                    pypi_0    pypi
cloudpickle               1.4.1                    pypi_0    pypi
codecov                   2.1.7                    pypi_0    pypi
coverage                  5.1                      pypi_0    pypi
decorator                 4.4.2                      py_0
distlib                   0.3.0                    pypi_0    pypi
docopt                    0.6.2                    pypi_0    pypi
etelemetry                0.2.1                    pypi_0    pypi
execnet                   1.7.1                    pypi_0    pypi
filelock                  3.0.12                   pypi_0    pypi
frozendict                1.2                      pypi_0    pypi
identify                  1.4.20                   pypi_0    pypi
idna                      2.9                      pypi_0    pypi
ipython                   7.15.0                   py38_0
ipython_genutils          0.2.0                    py38_0
jedi                      0.17.1                   py38_0
jsonschema                3.2.0                    pypi_0    pypi
libcxx                    10.0.0                        1
libedit                   3.1.20191231         haf1e3a3_0
libffi                    3.3                  h0a44026_1
lxml                      4.5.1                    pypi_0    pypi
mock                      4.0.2                    pypi_0    pypi
more-itertools            8.4.0                    pypi_0    pypi
ncurses                   6.2                  h0a44026_1
nexus-sdk                 0.3.2                    pypi_0    pypi
nodeenv                   1.4.0                    pypi_0    pypi
numpy                     1.19.0                   pypi_0    pypi
openssl                   1.1.1g               h1de35cc_0
packaging                 20.4                     pypi_0    pypi
parso                     0.7.0                      py_0
pathspec                  0.8.0                    pypi_0    pypi
pexpect                   4.8.0                    py38_0
pickleshare               0.7.5                 py38_1000
pip                       20.1.1                   py38_1
pluggy                    0.13.1                   pypi_0    pypi
pre-commit                2.5.1                    pypi_0    pypi
prompt-toolkit            3.0.5                      py_0
psutil                    5.7.0                    pypi_0    pypi
ptyprocess                0.6.0                    py38_0
puremagic                 1.10                     pypi_0    pypi
py                        1.9.0                    pypi_0    pypi
pydra                     0.6.1+10.g83c1b78           dev_0    <develop>
pygments                  2.6.1                      py_0
pyld                      2.0.2                    pypi_0    pypi
pyparsing                 2.4.7                    pypi_0    pypi
pyrsistent                0.16.0                   pypi_0    pypi
pytest                    5.4.3                    pypi_0    pypi
pytest-cov                2.10.0                   pypi_0    pypi
pytest-env                0.6.2                    pypi_0    pypi
pytest-forked             1.2.0                    pypi_0    pypi
pytest-rerunfailures      9.0                      pypi_0    pypi
pytest-xdist              1.32.0                   pypi_0    pypi
python                    3.8.3                h26836e1_1
python-dateutil           2.8.1                    pypi_0    pypi
pyyaml                    5.3.1                    pypi_0    pypi
readline                  8.0                  h1de35cc_0
regex                     2020.6.8                 pypi_0    pypi
requests                  2.24.0                   pypi_0    pypi
setuptools                47.3.1                   py38_0
simplejson                3.17.0                   pypi_0    pypi
six                       1.15.0                     py_0
sqlite                    3.32.3               hffcf06c_0
sseclient                 0.0.26                   pypi_0    pypi
tabulate                  0.8.7                    pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
tk                        8.6.10               hb0a8c7a_0
toml                      0.10.1                   pypi_0    pypi
tornado                   6.0.4                    pypi_0    pypi
traitlets                 4.3.3                    py38_0
typed-ast                 1.4.1                    pypi_0    pypi
urllib3                   1.25.9                   pypi_0    pypi
virtualenv                20.0.25                  pypi_0    pypi
wcwidth                   0.2.4                      py_0
wheel                     0.34.2                   py38_0
xz                        5.2.5                h1de35cc_0
zlib                      1.2.11               h1de35cc_3
```

I'd love to have this fixed .... somehow .... that doesn't include nuking my laptop.

**Comment by satra** [comment]
can we try redoing your conda setup?

1. new setup complete isolation. remove old conda from your bashrc/path (comment it out). start a new terminal. download and install miniconda in a new location. add this new location to your path. repeat the install steps above.

2. update your base environment. this depends on how old your conda is. and repeat the above steps.

also always useful to do `which -a python` and `which -a conda` to know how many different routes to python libraries you could potentially be dealing with.

**Comment by axiezai** [comment]
I will try a fresh conda after wrapping up a project this week!

**Comment by effigies** [comment]
You'll also want to resync changes in https://github.com/nipype/pydra-tasks-template/blob/master/setup.cfg. There were a couple issues that needed fixing to make namespace packages work correctly: https://github.com/nipype/pydra-tasks-template/commit/1afed647abc18acbc8bc5e6d0d1a4a951403538e

**Comment by axiezai** [comment]
I am positive I made those changes before running into this problem, but I will double check when I recreate my conda environment! Thank you!

**Comment by effigies** [comment]
Ugh. I'm having this issue now too.

**Comment by axiezai** [comment]
Closing this issue since the namespace was fixed with 288e2276a96b1dbbaaab8c6d0e7c03893f798ee0 and implemented here with dd64a44592df396437ba0bcb8f9c099b6f35be29
